test/integration/conftest.py

```diff
@@ -0,0 +1,8 @@
+def pytest_unconfigure(config):
+    try:
+        # This needs to be run if no test were run.
```

Comment:
That doesn't look quite right but is probably necessary because these tests are a mix of unittest and pytest. Normally all fixtures should cleanup after themselves.

Reply:
We do need it, I think (and this one should be pure pytest): if you keep that test, but don't pass any data, then the fixture will be called, but no tests will be generated, and the unconfigure hook will clean up. Remove the hook, and the fixture will keep running until you kill it manually.